src/tree.rs

```diff
@@ -11,8 +13,15 @@ use crate::{
 /// Generally, Taffy expects your Node tree to be indexable by stable indices. A "stable" index means that the Node's ID
 /// remains the same between re-layouts.
 pub trait LayoutTree {
+    type ChildIter<'a>: Iterator<Item = &'a DefaultKey>
+    where
+        Self: 'a;
+
     /// Get the list of children IDs for the given node
-    fn children(&self, node: Node) -> &[Node];
+    fn children<'a>(&'a self, node: Node) -> Self::ChildIter<'a>;
+
+    /// Get the number of children for the given node
+    fn num_children(&self, node: Node) -> usize;
```

Comment:
This is probably highly subjective, but what would you think about renaming this to something like `child_count`?

Reply:
Yeh sure, I'm not too fussy on the name so I'm happy to change it to `child_count` if that's preferable.

Reply:
Personally I like it more because there are less words "omitted", but I'm also happy to hear other opinions on this :)

Also not too important